lms/djangoapps/experiments/utils.py

```diff
@@ -78,15 +78,29 @@ def check_and_get_upgrade_link_and_date(user, enrollment=None, course=None):
         if course is None:
             course = enrollment.course
         elif enrollment.course_id != course.id:
-            logger.warn(u'{} refers to a different course than {} which was supplied'.format(
-                enrollment, course
-            ))
+            logger.warn(u'{} refers to a different course than {} which was supplied. Enrollment course id={}, repr={},'
```

Comment:
Nit: When calling logging functions (`.warn`, `.error`, etc), it's better to use the old-style formatting string, and pass values to be formatted in as arguments to the function. So, rather than `.warn("The error: {}".format(error))`, prefer `.warn("The error: %s", error)`. This lets the logging infrastructure access those values during processing (if needed).

That said, it likely doesn't make a difference in this particular case.

Reply:
Also, style tip: when you're doing new-style formatting, you can use the `!r` modifier to format as a `repr`. So, for you could use something like `repr={!r}` and then not have to call `repr` yourself on the values.